packages/flutter_tools/test/general.shard/build_system/targets/assets_test.dart

```diff
@@ -97,6 +97,26 @@ flutter:
     ProcessManager: () => FakeProcessManager.any(),
     Platform: () => platform,
   });
+
+  testUsingContext('Throws exception if pubspec contains missing files', () async {
+    fileSystem.file('pubspec.yaml')
+      ..createSync()
+      ..writeAsStringSync('''
+name: example
+
+flutter:
+  assets:
+    - assets/foo/bar2.png
+
+''');
+
+    expect(() async => await const CopyAssets().build(environment),
+      throwsA(isA<Exception>()));
```

Comment:
qq: is it worth checking the exception message as well?

Reply:
The exception is pretty nothingburger, the meaningful message is over stderr from checks like https://github.com/flutter/flutter/blob/master/packages/flutter_tools/lib/src/flutter_manifest.dart#L206